src/SelectMenu/SelectMenuTab.js

```diff
@@ -0,0 +1,41 @@
+import React, {useContext} from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import {MenuContext} from './SelectMenuContext'
+import {tabStyles} from './SelectMenuStyles'
+
+const TabBase = ({tabName, index, onClick, ...rest}) => {
+  const menuContext = useContext(MenuContext)
+  const handleClick = e => {
+    // if consumer has attached an onClick event, call it
+    onClick && onClick(e)
+    if (!e.defaultPrevented) {
+      menuContext.setSelectedTab(tabName)
+    }
+  }
+
+  // if no tab is selected when the component renders, show the first tab
+  if (!menuContext.selectedTab && index === 0) {
+    menuContext.setSelectedTab(tabName)
```

Comment:
I think this is going to cause issues... You'll trigger another render inside a render, and typically React complains (loudly) in the console if you do this.

Reply:
Probably want to use `useEffect(() => {}, [])`